File: Python/silnia.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-
#  n! = 1 dla {0,1}
#  n! = 1 * 2 * ... * n dla N+ - {0, 1}

#  n! = (n-1)! * n
#  4! = 3! * 4


def silnia_rek(n):
    if n < 2:
        return 1
    return silnia_rek(n - 1) * n


def silnia_it(n):
    """Funkcja oblicza  iteracyjnie silnię  liczby naturalnej"""
    wynik = 1
    for i in range(2, n + 1):
        wynik = wynik * i
    return wynik


def main(args):
    a = int(input("Podaj liczbę: "))
    assert type(a) == int

    assert silnia_it(0) == 1
    assert silnia_it(1) == 1
    assert silnia_it(7) == 5040
    assert silnia_rek(7) == 5040
    print("Silnia: ", silnia_rek(a))

    return 0


if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))

File: cpp/silnia.cpp
/*
 * silnia.cpp
 */


#include <iostream>
using namespace std;

int silnia(int(n))
	{
		int wynik = 1;
		for (int i = 2; i<=n ; i++)
		{
			wynik = wynik * i;	
			cout << wynik << endl;
		}	
		return wynik;
	}
int silnia_rek(int n)
{
    if (n < 2)
        return 1;
    else
        return silnia_rek(n- 1) * n;
}

int main(int argc, char **argv)
{
	int n = 1;
	cout << "Podaj liczbe " << endl;
	cin >> n ;
	cout << "Silnia wynosi: " << silnia(int(n)) << endl;
	return 0;
}

